### DISCUSSION TITLE
manpage jail.conf

### DISCUSSION THREAD

**Discussion by averlon** [fail2ban/fail2ban#3559] [General]
Hi,
I have had a look at the manpage for jail.conf and was looking for the parameter "port" for a jail, but I was not able to find it. At least, I did not recognize, probably I have overseen it.

In addition, in some configuration e.g. the TIME ABBREVIATION FORMAT "10m" is used.

In the manpage explanations I have found "minutes?, min?, mm?" but no "m?". It looks like, that also "h" works, but this is also not listed. At least as I do understand the description.

Not sure if I am not wrong!

**Comment by sebres** [comment]
> was looking for the parameter "port" for a jail, but I was not able to find it. At least, I did not recognize, probably I have overseen it

You didn't, it is not described in man page as many other interpolation parameters that explained in configuration only. This parameter will be used in oneport or multiport banning actions only and supplied to the action as it is:
https://github.com/fail2ban/fail2ban/blob/bcaf1e714e7a9006677f841af0f3802a3b5419ca/config/jail.conf#L194-L196
...
https://github.com/fail2ban/fail2ban/blob/bcaf1e714e7a9006677f841af0f3802a3b5419ca/config/jail.conf#L212

The issue here is - this parameter must be supplied in format (numeric or not, port range, etc) the banning action will support that, so this is dependent on the action one use for the banning. Allports actions don't need it at all.

Anyway there are many parameters that are described in the comment line directly in `.conf` file only. They are not directly fail2ban parameters, but rather used as some substitution inside the configuration only.

> In the manpage explanations I have found "minutes?, min?, mm?" but no "m?". 

Because `?` in `mm?` means that second `m` is optional (so either `mm` or `m`). And `m?` would mean either `m` or empty.

This is what the documentation says:

*The question mark (?) means the optional character, so `day` as well as `days` can be used.*

> It looks like, that also "h" works, but this is also not listed. 

It is the same: `hh?` means either `hh` or `h`.
